src/main/java/mandel/dictionary/EnglishDictionary.java

```diff
@@ -0,0 +1,46 @@
+package mandel.dictionary;
+
+import com.opencsv.CSVReader;
+import com.opencsv.exceptions.CsvValidationException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.*;
+
+/**
+ * Reads the englishDictionary file ONCE.
+ */
+public class EnglishDictionary {
+    List<String[]> wordAndDef = new ArrayList<>();
+
+    public EnglishDictionary() throws CsvValidationException, IOException {
+        // gets the file from the "resources" directory
+        InputStream in = EnglishDictionary.class.getResourceAsStream(
+                "/englishDictionary.csv");
+
+        CSVReader reader = new CSVReader(new InputStreamReader(in));
+        String[] record;
+
+        while ((record = reader.readNext()) != null) {
+            // record is ONE line of the CSV
+            wordAndDef.add(record);
+        }
+        reader.close();
+    }
+
+    /**
+     * @param word to look up.
+     * @return a List of definitions, or an empty list if the word doesn't exist.
+     */
+    public List<String> getDefinition(String word) {
+        List<String> allDefs = new ArrayList<>();
+
+        for (String[] record : wordAndDef) {
+            if (record[0].equalsIgnoreCase(word)) {
+                allDefs.addAll(Arrays.asList(record).subList(1, record.length));
```

Comment:
How does this work?

Reply:
The foreach loop goes through every record (or line) that's stored in the wordsAndDef list - the list contains all the words and their definitions from the csv file. The if statement checks if the first index of record (the word to translate) is equal to the word that's passed into the method - the ignore case is just in case a word that starts with a lowercase letter gets passed in, so that word still gets translated. The last line works like a regular for loop that starts from 1 and goes until the end of the record. This loop starts from after the word to be translated and adds the translation of the word to the allDefs list which in the end will contain the full list of translations for the given word. (I actually wrote this line out as a for loop from i=1 to record.length and added record[i] to the allDefs list, but intellij consolidated it for me into an addAll()).

Reply:
If there are 2 definitions for a word, then how many elements will be in allDefs?
How many elements are in record.subList(1, record.length)?

Reply:
If there are 2 definitions for a word there are 4 elements in allDefs. For each word, there are 2 elements in  record.subList(1, record.length) -- the type of the word (ex: noun) and the actual definition.

Reply:
I never asked to return the type of word, only the definition.